### PR TITLE
remove omitempty tag from PathType

### PR DESCRIPTION
**Pull request by sschne**
#### What type of PR is this?
/kind cleanup

#### What this PR does / why we need it:
The Ingress API in networking.k8s.io/v1 makes the PathType a required argument. Removing the omitempty struct tag reflects this change and updates the swagger document to adds the property to the list of required properties.

#### Which issue(s) this PR fixes:

Fixes #

#### Does this PR introduce a user-facing change?

```release-note
NONE
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs

```

### CODE REVIEW

**Comment by sschne** [comment]
I'm unsure, if this change qualifies for an API review, it says
> Any change to the meaning, validation, or behavior of a field
which actually does, but this is just a fix for an already known API change. If anyone could help me out here?

File: staging/src/k8s.io/api/networking/v1/types.go [hunk 1]
```diff
@@ -426,7 +426,7 @@ type HTTPIngressPath struct {
 	//   the IngressClass. Implementations can treat this as a separate PathType
 	//   or treat it identically to Prefix or Exact path types.
 	// Implementations are required to support all path types.
-	PathType *PathType `json:"pathType,omitempty" protobuf:"bytes,3,opt,name=pathType"`
+	PathType *PathType `json:"pathType" protobuf:"bytes,3,opt,name=pathType"`
```
Comment (robscott) [inline staging/src/k8s.io/api/networking/v1/types.go:429]: If we're removing omitempty shouldn't this type also change to a pointer? I'm not sure if that's backwards compatible though.

**Comment by thockin** [comment]
Who says it is required?  It still has a default and as such is strictly optional.  I think we should NOT do this, unless we have a really good reason.

**Comment by sschne** [comment]
> Who says it is required? It still has a default and as such is strictly optional. I think we should NOT do this, unless we have a really good reason.

According to the Changelog introducing Ingress in networking.k8s.io/v1, the pathType no longer has a default value compared to networking.ik8s.io/v1beta1, see https://github.com/kubernetes/kubernetes/pull/89778

**Comment by aojea** [comment]
> According to the Changelog introducing Ingress in networking.k8s.io/v1, the pathType no longer has a default value compared to networking.k8s.io/v1beta1, see #89778

 ... but the field can not be empty
https://github.com/aojea/kubernetes/blob/2eb50ee2fd5e5c1db64121359bd5014552ba7858/pkg/apis/networking/validation/validation.go#L390-L392

**Review by thockin — COMMENTED** [review]
I see.  It does set defaults in beta and not v1.  Then yeah, this can be a non-pointer non-omitempty.  Conversion code has to handle that, and since it has to round-trip to v1b1, it's a little messy and manual to go from pointer to non-pointer.  So leaving it pointer seems OK.

/approve

back to @robscott

**Comment by robscott** [comment]
Thanks!

/lgtm
/ok-to-test

**Pull request merged**